File: spark-submitter/src/main/java/dk/sdu/datascience/kafka/structure/schemas/Status.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package dk.sdu.datascience.kafka.structure.schemas;

import com.google.gson.annotations.SerializedName;

/**
 *
 * @author Async api generated
 */
public class Status {
	public enum eventEnum {
		@SerializedName("STARTING")
			STARTING,
		@SerializedName("DONE")
			DONE,
		@SerializedName("STOPPED")
			STOPPED,
	}
	
    private eventEnum event;

    /**
     * @return eventEnum
     */
    public eventEnum getEvent() {
        return event;
    }

    /**
     * @param event to set
     */
    public void setEvent(eventEnum event) {
        this.event = event;
    }
}

File: Visualizer/kafka/schemas/Status.js

const eventEnum = Object.freeze({
			"STARTING": "STARTING",
			"DONE": "DONE",
			"STOPPED": "STOPPED",
    })
module.exports.eventEnum = eventEnum;
module.exports = class Status {
    constructor(){

    }
    
    /**
    *
    * @param { string } event
    */
    setData(
        event
    ){
        this.event=event;
    }


    
    /**
    * Copy a js object into this.
    * @param {*} jsonObject the js object 
    */
    copyInto(jsonObject){
        if(jsonObject.event){
            this.event=jsonObject.event;
        }
    }
}
